packages/webapp/src/components/Map/Modals/BulkSensorUploadModal/useValidateBulkSensorData.js

```diff
@@ -0,0 +1,219 @@
+      /* eslint-disable-next-line */
+      mask: /^(\+|-)?(?:180(?:(?:\.0{1,6})?)|(?:[0-9]|[1-9][0-9]|1[0-7][0-9])(?:(?:\.[0-9]{1,30})?))$/,
+      columnName: SENSOR_LONGITUDE,
+    },
+    {
+      errorMessage: t('FARM_MAP.BULK_UPLOAD_SENSORS.VALIDATION.SENSOR_READING_TYPES'),
+      /* eslint-disable-next-line */
+      mask: /^\s*(?:\w+\s*,\s*){2,}(?:\w+\s*)$/,
+      columnName: SENSOR_READING_TYPES,
+      validate(rowNumber, columnName, value) {
+        if (typeof value !== 'string') return;
+        const inputReadingTypes = value.split(',');
+        if (!inputReadingTypes.length) return;
+        const invalidReadingTypes = inputReadingTypes.reduce((acc, fieldName) => {
+          if (!requiredReadingTypes.includes(fieldName.trim())) {
+            acc.push(fieldName.trim());
+          }
+          return acc;
+        }, []);
+        if (!invalidReadingTypes.length) return;
+        return {
+          row: rowNumber,
+          column: columnName,
+          errorMessage: t('FARM_MAP.BULK_UPLOAD_SENSORS.VALIDATION.SENSOR_READING_TYPES'),
+          value: invalidReadingTypes.join(','),
+        };
+      },
+    },
+  ];
+
+  useEffect(() => {
+    if (!disabled) setDisabled(0);
+    else setDisabled(bulkSensorsUploadResponse.loading ? -1 : 1);
+  }, [bulkSensorsUploadResponse?.loading]);
+
+  const validateExcel = (rows) => {
+    let errors = [];
+    for (let i = 0; i < rows.length; i++) {
+      let element = rows[i];
+      for (const validationField of validationFields) {
+        const COLUMN = validationField.columnName;
+        const validColumn = validationField.mask.test(element[COLUMN]);
+        if (!validColumn) {
+          errors.push({
+            row: i + 2,
+            column: COLUMN,
+            errorMessage: validationField.type,
+            value: element[COLUMN],
+          });
+        } else {
+          // find for other errors after regex check.
+          if (validationField.hasOwnProperty('validate')) {
+            const validationError = validationField.validate(i + 2, COLUMN, element[COLUMN]);
+            if (validationError) {
+              errors.push(validationError);
+            }
+          }
+        }
+      }
+    }
+    return errors;
+  };
+
+  const onUploadClicked = async (e) => {
+    e.preventDefault();
+    const file = fileInputRef.current.files[0];
+    if (file) {
+      onUpload(file);
+    }
+  };
+
+  const checkRequiredColumnsArePresent = (sensorObject = {}) => {
+    const missingColumns = requiredFields.filter(
+      (fieldName) => !Object.keys(sensorObject).includes(fieldName),
+    );
+    return missingColumns.length
+      ? [
+          {
+            row: 1,
+            column: missingColumns.join(','),
+            errorMessage: t('FARM_MAP.BULK_UPLOAD_SENSORS.VALIDATION.MISSING_COLUMNS'),
+            value: '',
+          },
+        ]
+      : [];
+  };
+
+  const handleSelectedFile = async (e) => {
+    const file = e.target.files[0];
+    if (file) {
```

Comment:
I prefer "early return" to deeper nesting.

Reply:
@anant00729